pytensor/tensor/math.py

```diff
@@ -1571,6 +1574,38 @@ def std(input, axis=None, ddof=0, keepdims=False, corrected=False):
     return ret
 
 
+def median(input, axis=None):
+    """
+    Computes the median along the given axis(es) of a tensor `input`.
+
+    Parameters
+    ----------
+    axis: None or int or (list of int) (see `Sum`)
+        Compute the median along this axis of the tensor.
+        None means all axes (like numpy).
+
+    Notes
+    -----
+    This function uses the numpy implementation of median.
+    """
+    from pytensor.ifelse import ifelse
+
+    input = as_tensor_variable(input)
+    sorted_input = input.sort(axis=axis)
+    shape = input.shape[axis]
```

Comment:
This will fail for axis=None

Reply:
Can we do this?
```
if axis is None:
        input = input.flatten()
        axis=0
```
The output matches the expected output in case of `np.median`.
In other places whenever axis=None, we use ```axis = list(range(input.ndim))```. However the ```take``` does not allow axis to be a list and only works if axis is integer or None. 
Tested on:
```
([1, 7, 3, 6, 5, 2, 4], None),
median = 5

[[[6, 2, 3], [1, 5, 8], [4, 7, 9]], [[5, 3, 4], [8, 6, 2], [7, 1, 9]]], None),
median = 5.0 ( using both our logic here and numpy)

```

Reply:
Yes, we can do that. But we need something a bit different for partial axis like `median(x, axis=(1, 2)` for a 3d tensor. In that case we need to transpose -> reshape -> median logic -> transpose again (I think). If you play a bit with the numpy function you'll see what it is doing